lib/project.rb

```diff
@@ -161,15 +173,28 @@ def init_workspace
         @logger.warn('Workspace path already exists')
       end
       @logger.info("Workspace path: #{@workspace_path}")
+    end
+
+    def restore_session
+      @workspace_path = @options.test.session
 
+      raise AutoHCKError, 'Workspace path does not exist could not load session' unless File.directory?(@workspace_path)
+
+      @logger.info("Loading session from #{@workspace_path}")
+    end
+
+    def save_session
+      Session.save(@workspace_path, @options, @logger)
+    end
+
+    def add_latest_symlink
       begin
         File.delete("#{@config['workspace_path']}/latest")
       rescue Errno::ENOENT
         # firts run, no symlink to delete
       end
 
       File.symlink(@workspace_path, "#{@config['workspace_path']}/latest")
-      @setup_manager_type&.enter @workspace_path
     end
```

Comment:
I suggest merging this into `create_workspace`. Creating a workspace automatically makes it the latest so it makes sense to automatically update the symlink. The code in `init_workspace` will be more natural as you can write it to say "if being restored, restore the workspace; otherwise create a new one".